backend/common/ai_drive.py

```diff
@@ -54,37 +50,47 @@ def dl_tabular_drive(
         epochs (int, optional): number of epochs/rounds to run model on
         shuffle (bool, optional): should the dataset be shuffled prior to train/test split
     """
-    target = params.get("target", None)
+    params = trainspace_data.parameters_data
+
+    target = params.get("target_col", None)
     features = params.get("features", None)
     problem_type = params["problem_type"]
     optimizer_name = params["optimizer_name"]
     criterion = params["criterion"]
-    default = params.get("default", None)
+    default = (
+        trainspace_data.dataset_data["name"]
+        if trainspace_data.dataset_data.get("is_default_dataset")
+        else None
+    )
     epochs = params.get("epochs", 5)
     shuffle = params.get("shuffle", True)
     test_size = params.get("test_size", 0.2)
     batch_size = params.get("batch_size", 20)
 
+    user_arch = params["layers"]
+
     category_list = []
-    if not default:
-        if fileURL:
-            read_dataset(fileURL)
-        elif json_csv_data_str:
-            pass
-        else:
-            raise ValueError("Need a file input")
 
     if default and problem_type.upper() == "CLASSIFICATION":
         X, y, category_list = get_default_dataset(default.upper(), target, features)
     elif default and problem_type.upper() == "REGRESSION":
         X, y, discard = get_default_dataset(default.upper(), target, features)
```

Comment:
what does `discard` do here?

this variable is the only difference between `get_default_dataset` for regression v. classification